crates/gitbutler-tauri/src/repo.rs

```diff
@@ -92,4 +96,61 @@ pub mod commands {
         let project = projects.get(project_id)?;
         Ok(project.read_file_from_workspace(relative_path)?)
     }
+
+    #[tauri::command(async)]
+    #[instrument(skip(projects))]
+    pub fn pre_commit_hook(
+        projects: State<'_, projects::Controller>,
+        project_id: ProjectId,
+        ownership: BranchOwnershipClaims,
+    ) -> Result<HookResult, Error> {
+        let project = projects.get(project_id)?;
+        let ctx = CommandContext::open(&project)?;
+        let repo = ctx.repo();
+        let diffs = gitbutler_diff::workdir(
+            ctx.repo(),
+            repo.head()
+                .context("no head")?
+                .peel_to_commit()
+                .context("no commit")?
+                .id(),
+        )?;
+        let selected_files = staging::filter_diff_by_hunk_ids(
+            diffs,
+            ownership
+                .claims
+                .iter()
+                .map(|claim| {
+                    (
+                        claim.file_path.clone(),
+                        claim.hunks.iter().map(HunkId::from).collect(),
+                    )
+                })
+                .collect(),
```

Comment:
You can make a function in gitbutler-branch-actions so that you can test this and also to keep the tauri crate lean

Reply:
Done!